cli/command/trust/sign.go

```diff
@@ -50,12 +57,41 @@ func signImage(cli command.Cli, image, tag string) error {
 			fmt.Fprintf(cli.Out(), "Finished initializing %q\n", notaryRepo.GetGUN().String())
 		default:
 			return trust.NotaryError(repoInfo.Name.Name(), err)
+
 		}
 	}
-
-	// TODO: craft and sign the target using image.addTargetToAllSignableRoles (needs to be exported)
+	ctx := context.Background()
+	requestPrivilege := command.RegistryAuthenticationPrivilegedFunc(cli, repoInfo.Index, "push")
+	target, err := createTarget(notaryRepo, tag)
+	if err != nil && strings.Contains(err.Error(), "No valid trust data for") {
+		image.TrustedPush(ctx, cli, repoInfo, ref, *authConfig, requestPrivilege)
```

Comment:
we should check the error from `TrustedPush`, this can probably simplified to:

`return image.TrustedPush(ctx, cli, repoInfo, ref, *authConfig, requestPrivilege)`